src/mame/misc/astrcorp.cpp

```diff
@@ -1857,82 +2377,63 @@ const astoneage_state::decryption_info astoneage_state::dinodino_table = {
-	rom[0x004/2] = 0x0000;
-	rom[0x006/2] = 0x0446;
+	u16 * const rom = (u16 *)memregion("maincpu")->base();
+	rom[0x00004/2] = 0x0000;
+	rom[0x00006/2] = 0x0440;
 
-	// enable the patch below to pass ROM checksum, but then it jumps into the weeds after expecting to read something from 0xb80000
-	//rom[0x30b70/2] = 0xe5f5;
-}
+	rom[0x00400/2] = 0x4e75; // overlay!?
 
-void magibomb_state::init_magibomb()
-{
-	// decrypt data
-	u8 *rom = memregion("maincpu")->base();
+	rom[0x02872/2] = 0x4e75; // Mirror ROM word checksum (it expects 0)
+#endif
 
-	for (u32 i = 1; i < 0x20000; i += 2)
-	{
-		rom[i] = bitswap<8>(rom[i], 6, 7, 5, 0, 4, 1, 3, 2);
-	}
+	interleave_sprites_16x32();
+}
 
-	// descramble addresses
-	std::unique_ptr<u8[]> tmp = std::make_unique<u8[]>(0x20000);
+void astoneag_state::interleave_sprites_16x32()
+{
+	u8 * const rom = memregion("sprites")->base();
+	const u32 size = memregion("sprites")->bytes();
 
-	for(u32 i = 0; i < 0x20000; i++)
-	{
-		tmp[i] = rom[i];
-	}
+	std::unique_ptr<u8[]> tmp = std::make_unique<u8[]>(size);
+	std::copy_n(rom, size, &tmp[0]);
 
-	for(u32 i = 0; i < 0x20000; i++)
-	{
-		u32 dest =
-			(i & 0xffff20ff) |
-			(BIT(i, 14)  << 15) |
-			(BIT(i, 15)  << 14) |
-			(BIT(i, 8)   << 12) |
-			(BIT(i, 12)  << 11) |
-			(BIT(i, 9)   << 10) |
-			(BIT(i, 11)  <<  9) |
-			(BIT(i, 10)  <<  8);
-		rom[dest] = tmp[i];
-	}
+	for (u32 i = 0; i != size; i += 1)
+		rom[i] = tmp[(i & ~0xff) | (bitswap<8>(i, 7,6,5, 3,2,1, 4, 0) ^ 2)];
 }
 
 } // Anonymous namespace
 
-
-GAME( 2000,  showhand,  0,        showhand,  showhand, astrocorp_state, init_showhand,  ROT0, "Astro Corp.",        "Show Hand (Italy)",                  MACHINE_SUPPORTS_SAVE )
-GAME( 2000,  showhanc,  showhand, showhanc,  showhanc, astrocorp_state, init_showhanc,  ROT0, "Astro Corp.",        "Wang Pai Dui Jue (China)",           MACHINE_SUPPORTS_SAVE )
-GAME( 2002,  skilldrp,  0,        skilldrp,  skilldrp, astrocorp_state, empty_init,     ROT0, "Astro Corp.",        "Skill Drop Georgia (Ver. G1.0S)",    MACHINE_SUPPORTS_SAVE )
-GAME( 2003,  speeddrp,  0,        speeddrp,  skilldrp, astrocorp_state, empty_init,     ROT0, "Astro Corp.",        "Speed Drop (Ver. 1.06)",             MACHINE_SUPPORTS_SAVE )
-
-// Encrypted games (not working):
+//     YEAR   NAME       PARENT    MACHINE    INPUTS     STATE            INIT            ROT   COMPANY        FULLNAME                                         FLAGS                                                  LAYOUT
+GAMEL( 2000,  showhand,  0,        showhand,  showhand,  astrocorp_state, init_showhand,  ROT0, "Astro Corp.", "Show Hand (Italy)",                             MACHINE_SUPPORTS_SAVE,                                 layout_showhand  )
+GAMEL( 2000,  showhanc,  showhand, showhanc,  showhanc,  astrocorp_state, init_showhanc,  ROT0, "Astro Corp.", "Wang Pai Dui Jue (China)",                      MACHINE_SUPPORTS_SAVE,                                 layout_showhanc  )
```

Comment:
Please use “Wangpai Duijue” – don’t put spaces between every syllable in Pinyin.

Reply:
Done.